File: Suite de Conway - Look and Say/conway.js
//  Créer une fonction decritChainequi prend en paramètre une string et
// renvoie une string qui décrit les caractères qui constituent la chaîne en paramètre.

function describeString(myString) {
  // définition d'un tableau contenant les chaînes de caractères identiques
  let similarCaracs = myString[0];
  for (i = 1; i < myString.length; i++) {
    if (myString[i] == myString[i - 1]) {
      similarCaracs += myString[i];
    } else {
      similarCaracs += " " + myString[i];
    }
  }
  let caracsArray = similarCaracs.split(" ");
  // comptage des caractères identiques dans chaque élément du tableau
  let describeResult = "";
  for (j = 0; j < caracsArray.length; j++) {
    // console.log(caracsArray[j]);
    describeResult += caracsArray[j].length + caracsArray[j][0];
  }
  return describeResult;
}

// console.log(describeString("aabccd"));

// Créer une fonction suiteConway qui donne les n premiers termes de la suite
// qui commence par le caractère carac. n et carac sont passés en paramètres de la fonction.

function suiteConway(n, carac) {
  console.log(carac);
  let result = describeString(carac);
  console.log(result);
  for (k = 1; k < n - 1; k++) {
    console.log(describeString(result));
    result = describeString(result);
  }
}

// suiteConway(5, "a");
// suiteConway(10, "1");

// Afficher la suite de Conway générée dans un navigateur.
// Utiliser un texte centré pour l'afficher sous forme de pyramide.

function suiteConway2(n, carac) {
  let div = document.getElementById("suite");
  div.innerHTML = carac + "<br />";
  console.log(carac);

  let result = describeString(carac);
  div.innerHTML += result + "<br />";

  console.log(result);

  for (l = 1; l < n - 1; l++) {
    div.innerHTML += describeString(result) + "<br />";

    console.log(describeString(result));
    result = describeString(result);
  }
}

function getValue() {
  // Sélectionner l'élément input et récupérer sa valeur
  let numberInput = document.getElementById("numberInput").value;
  let caracInput = document.getElementById("caracInput").value;
  // Afficher la valeur
  //alert("vous avez saisi" + numberInput + caracInput);
  suiteConway2(numberInput, caracInput);
  suiteConway(numberInput, caracInput);
}

File: Suite de Conway - Look and Say/conway.py
# Créer une fonction decoupeChaine qui prend en paramètre une string
# et renvoie la même string dans laquelle les caractères successifs non identiques sont séparés par un espace.

def decoupe_chaine(string):
    res_string = string[0]
    for x in range(1, len(string)):
        if string[x] == string[x-1]:
            res_string += string[x]
        else:
            res_string += " " + string[x]
    return res_string


result1 = decoupe_chaine("aabb")
result2 = decoupe_chaine("abbccddd")
# print(result1, result2)

# Créer une fonction decritChaine, inspirée de decoupeChaine, qui prend en paramètre une string et
# renvoie une string qui décrit, tel qu'expliqué ci-dessus, les caractères qui constituent la chaîne en
# paramètre.


def decrit_chaine(string):
    res_string = string[0]
    for x in range(1, len(string)):
        if string[x] == string[x-1]:
            res_string += string[x]
        else:
            res_string += " " + string[x]
    res_list = res_string.split(' ')
    # print("tableau des résultats: ", res_list)
    res_def = ''
    for elem in res_list:
        res_def += str(len(elem)) + elem[0]
    return res_def


result3 = decrit_chaine('aabbc')

# print(result3)


# Créer une fonction suiteConway qui donne les n premiers termes de la suite
# qui commence par le caractère carac. n et carac sont passés en paramètres de la fonction.

def suiteConway(n, carac):
    print(carac)
    result = decrit_chaine(carac)
    print(result)
    for _ in range(n-2):
        print(decrit_chaine(result))
        result = decrit_chaine(result)


suiteConway(10, 'a')
suiteConway(4, '1')
